### PR TITLE
Fix ios_vlan to correctly identify unmodified config when having long interface names

### PR DESCRIPTION
**Pull request by pierremahot**
ios_vlan not working correctly #40142

##### SUMMARY
Change the command to get the interface in a vlan
"show vlan" => "show vlan brief"
Change the parsing of the return command of the switch.
The return of the ios command is fixed so i cut with fix number of carracter.
Adding looking for the next line to add the forgeted interfaces.
#40142

##### ISSUE TYPE
 - Bugfix Pull Request

##### COMPONENT NAME
Module network/ios/ios_vlan

##### ANSIBLE VERSION

```
ansible --version
ansible 2.5.2
  config file = /home/serveur/pmahot/ansible-skeleton/ansible.cfg
  configured module search path = [u'/home/serveur/pmahot/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/dist-packages/ansible
  executable location = /usr/bin/ansible
  python version = 2.7.9 (default, Jun 29 2016, 13:08:31) [GCC 4.9.2]
```

##### ADDITIONAL INFORMATION

```
- name: SWITCH vlan declaration
  delegate_to: "{{ switch }}"
  ios_vlan:
    vlan_id: "{{ vlan }}"
    name: "{{ name }}"
    state: present
  become: yes
```
Now working with real check of the config

### CODE REVIEW

**Comment by webknjaz** [comment]
-label needs_triage

File: lib/ansible/modules/network/ios/ios_vlan.py [hunk 1]
```diff
@@ -212,34 +212,44 @@ def map_params_to_obj(module):
...
 
     if not lines:
         return list()
-
     objs = list()
 
-    for l in lines:
-        splitted_line = l.strip().replace(",", "").split()
-        if splitted_line == []:
+    for index in range(len(lines)):
```
Comment (webknjaz) [inline lib/ansible/modules/network/ios/ios_vlan.py]: Please change this to 
```python
for current_line, next_line in zip(lines, lines[1:]):
```

File: lib/ansible/modules/network/ios/ios_vlan.py [hunk 2]
```diff
@@ -212,34 +212,44 @@ def map_params_to_obj(module):
...
+        vlanID = lines[index][0:4].strip().replace(",", "")
+        name = lines[index][5:37].strip().replace(",", "")
+        state = lines[index][38:47].strip().replace(",", "")
+        interfaces = lines[index][48:].strip().replace(",", "").split()
+        while ((index + 1) < len(lines)):
```
Comment (webknjaz) [inline lib/ansible/modules/network/ios/ios_vlan.py]: You won't need this complexity with the snippet I've posted above

File: lib/ansible/modules/network/ios/ios_vlan.py [hunk 3]
```diff
@@ -212,34 +212,44 @@ def map_params_to_obj(module):
...
+            if (len(interfaces_sup) > 0) and (len(vlanID_sup) < 1):
+                interfaces.extend(interfaces_sup)
+                index += 1
+	    else:
             break
-        obj = {}
-        obj['vlan_id'] = splitted_line[0]
-        obj['name'] = splitted_line[1]
-        obj['state'] = splitted_line[2]
+        if (vlanID == ""):
```
Comment (webknjaz) [inline lib/ansible/modules/network/ios/ios_vlan.py]: Please note that in Python we don't use brackets unless really needed.

File: lib/ansible/modules/network/ios/ios_vlan.py [hunk 4]
```diff
@@ -212,34 +212,44 @@ def map_params_to_obj(module):
...
+        obj = {}
+        obj['vlan_id'] = vlanID
+        obj['name'] = name
+        obj['state'] = state
         if obj['state'] == 'suspended':
             obj['state'] = 'suspend'
 
         obj['interfaces'] = []
-        if len(splitted_line) > 3:
+        if len(interfaces) > 0:
```
Comment (webknjaz) [inline lib/ansible/modules/network/ios/ios_vlan.py]: You can completely drop this conditional, since iterating over empty iterable produces the same effect.

File: lib/ansible/modules/network/ios/ios_vlan.py [hunk 5]
```diff
@@ -212,34 +212,44 @@ def map_params_to_obj(module):
...
+            continue
 
+        obj = {}
+        obj['vlan_id'] = vlanID
+        obj['name'] = name
+        obj['state'] = state
         if obj['state'] == 'suspended':
             obj['state'] = 'suspend'
 
         obj['interfaces'] = []
```
Comment (webknjaz) [inline lib/ansible/modules/network/ios/ios_vlan.py]: Actually, you can replace this dancing around if+for+append+extend (lines 245->250) with a simple list comprehension, which would look better readable:
```python
obj['interfaces'] = [i.replace('Gi', 'GigabitEthernet') for i in interfaces]
```
It will produce the very same result.

**Comment by webknjaz** [comment]
@pierremahot thank you for your contribution! I've added a few notes for you to fix as a review. Also for this to get accepted you need to make CI build green. Please refer to comments of @ansibot for more hints regarding this.

Besides that, we would really appreciate if you could extend integration tests with this case. This would (1) prove us and everyone concerned that the issue is really fixed and (2) prevent it from happening in future during refactoring or other kinds of changes.

File: lib/ansible/modules/network/ios/ios_vlan.py (hunk 2 above)
Comment (webknjaz) [reply lib/ansible/modules/network/ios/ios_vlan.py]: Trick credit: https://twitter.com/raymondh/status/967927989752098816

Comment (pierremahot) [reply lib/ansible/modules/network/ios/ios_vlan.py]: it can be on more than 2 lines in absolute so i don't see how to implement your sugestion (sorry i'm a begginer)

Comment (webknjaz) [reply lib/ansible/modules/network/ios/ios_vlan.py]: I've posted a snippet earlier, but it looks like github has folder my comment. Here it is: https://github.com/ansible/ansible/pull/40145#discussion_r188250854

File: lib/ansible/modules/network/ios/ios_vlan.py [hunk 6]
```diff
@@ -212,34 +212,38 @@ def map_params_to_obj(module):
...
-    for l in lines:
-        splitted_line = l.strip().replace(",", "").split()
-        if splitted_line == []:
-            break
-        obj = {}
-        obj['vlan_id'] = splitted_line[0]
-        obj['name'] = splitted_line[1]
-        obj['state'] = splitted_line[2]
+    for index in range(len(lines)):
+        vlanID = lines[index][0:4].strip().replace(",", "")
```
Comment (webknjaz) [inline lib/ansible/modules/network/ios/ios_vlan.py]: I'm not sure why you use exact index slices. Are they always this size?
I'd say that it's much safer to do `.split()` and that work with chunks:
```python
vlan_id, name, state, interfaces = [v.replace(',', '').strip() for v in current_line.split()]
```

please also note that in python we use so-called `snake_case` for variables, as opposed to `camelCase` (like in java or js)

**Review by webknjaz — COMMENTED** [review]
Please post a typical command output example here. We might be able to advise you a better way of parsing that.

File: lib/ansible/modules/network/ios/ios_vlan.py (hunk 2 above)
Comment (webknjaz) [reply lib/ansible/modules/network/ios/ios_vlan.py]: Oh.. I missed the part about "more than two lines", so please post example output, so we could take a closer look at the issue together :)

**Comment by pierremahot** [comment]
It looks like this
```
switch>terminal width 512
switch>sh vlan brief       
VLAN Name                             Status    Ports
---- -------------------------------- --------- -------------------------------
1    default                          active    Gi1/0/4, Gi1/0/5, Gi1/0/6, Gi1/0/7, Gi1/0/8, Gi1/0/9, Gi1/0/10, Gi1/0/11, Gi1/0/12, Gi1/0/13, Gi1/0/14, Gi1/0/15, Gi1/0/16, Gi1/0/17, Gi1/0/18, Gi1/0/19, Gi1/0/201
                                                Gi1/0/52
1002 fddi-default                     act/unsup 
1003 token-ring-default               act/unsup 
1004 fddinet-default                  act/unsup 
1005 trnet-default                    act/unsup 


```

**Comment by webknjaz** [comment]
@pierremahot how did it look without `brief`?

**Comment by webknjaz** [comment]
also, it `Ports` the only column, which can be multiline?

**Comment by pierremahot** [comment]
@webknjaz here thé différents
https://www.cisco.com/c/en/us/td/docs/ios/lanswitch/command/reference/lsw_book/lsw_s2.html

**Comment by pierremahot** [comment]
@webknjaz i think it's the only one but i am gooing to look

**Comment by webknjaz** [comment]
Do you have your own example of `show vlan` with multiple ports? It looks much more easily parsable to me, than `brief`

**Comment by webknjaz** [comment]
@pierremahot I've drafted an example parser for you. Feel free to reuse it:

`vlan_brief.out`:
```log
VLAN Name                             Status    Ports
---- -------------------------------- --------- -------------------------------
1    default                          active    Gi1/0/4, Gi1/0/5, Gi1/0/6, Gi1/0/7, Gi1/0/8, Gi1/0/9, Gi1/0/10, Gi1/0/11, Gi1/0/12, Gi1/0/13, Gi1/0/14, Gi1/0/15, Gi1/0/16, Gi1/0/17, Gi1/0/18, Gi1/0/19, Gi1/0/201
                                                Gi1/0/52
1002 fddi-default                     act/unsup
1003 token-ring-default               act/unsup
1004 fddinet-default                  act/unsup
1005 trnet-default                    act/unsup
```

`vlan_parser.py`:
```python
#! /usr/bin/env python

from pprint import pprint
import re


def read_out(fn):
    with open(fn) as f:
        return f.read()


def logical_rows(out):
    started_yielding = False
    cur_row = []
    for l in out.splitlines()[2:]:
        if not l:
            """Skip empty lines."""
            continue
        if '0' < l[0] < '9':
            """Line starting with a number."""
            if started_yielding:
                yield cur_row
                cur_row = []  # Reset it to hold a next chunk
            started_yielding = True
        cur_row.append(l)

    # Return the rest of it:
    yield cur_row


def ports_str_to_list(ports_str):
    return list(filter(bool, (p.strip() for p in ports_str.split(', '))))


def parse_logical_row(r):
    first_row = r[0]
    rest_rows = r[1:]
    vals = re.match(r'(?P<vlan>\d+)\s+(?P<name>[^\s]+)\s+(?P<status>[^\s]+)\s+(?P<ports>.*)', first_row).groupdict()
    vals['ports'] = ports_str_to_list(vals['ports'])
    vals['ports'].extend(prts_r for prts in rest_rows for prts_r in ports_str_to_list(prts))
    #print(vals['vlan'], vals['name'], vals['status'], vals['ports'])
    return vals


def parse_vlan_brief():
    vlan_out = read_out('vlan_brief.out')
    for r in logical_rows(vlan_out):
        row_object = parse_logical_row(r)
        pprint(row_object)  # Outputs parsed dict per row


__name__ == '__main__' and parse_vlan_brief()
```

it works:
```shell
$ ./vlan_parser.py
{'name': 'default',
 'ports': ['Gi1/0/4',
           'Gi1/0/5',
           'Gi1/0/6',
           'Gi1/0/7',
           'Gi1/0/8',
           'Gi1/0/9',
           'Gi1/0/10',
           'Gi1/0/11',
           'Gi1/0/12',
           'Gi1/0/13',
           'Gi1/0/14',
           'Gi1/0/15',
           'Gi1/0/16',
           'Gi1/0/17',
           'Gi1/0/18',
           'Gi1/0/19',
           'Gi1/0/201',
           'Gi1/0/52'],
 'status': 'active',
 'vlan': '1'}
{'name': 'fddi-default', 'ports': [], 'status': 'act/unsup', 'vlan': '1002'}
{'name': 'token-ring-default',
 'ports': [],
 'status': 'act/unsup',
 'vlan': '1003'}
{'name': 'fddinet-default', 'ports': [], 'status': 'act/unsup', 'vlan': '1004'}
{'name': 'trnet-default', 'ports': [], 'status': 'act/unsup', 'vlan': '1005'}
```

File: test/units/modules/network/ios/test_ios_vlan.py [hunk 7]
```diff
@@ -0,0 +1,55 @@
+# (c) 2016 Red Hat Inc.
```
Comment (webknjaz) [inline test/units/modules/network/ios/test_ios_vlan.py]: `+ two years

**Comment by pierremahot** [comment]
@webknjaz the out of the show vlan brief is exactly the same of the begining of the show vlan command but it's only with more information unecessary to the function

File: lib/ansible/modules/network/ios/ios_vlan.py [hunk 8]
```diff
@@ -212,34 +212,38 @@ def map_params_to_obj(module):
...
+    output = run_commands(module, ['show vlan brief'])
     lines = output[0].strip().splitlines()[2:-1]
 
     if not lines:
         return list()
-
     objs = list()
 
-    for l in lines:
-        splitted_line = l.strip().replace(",", "").split()
```
Comment (webknjaz) [inline lib/ansible/modules/network/ios/ios_vlan.py]: @pierremahot please reuse my snippet to have better correct parser.

File: lib/ansible/modules/network/ios/ios_vlan.py [hunk 9]
```diff
@@ -211,41 +211,48 @@ def map_params_to_obj(module):
...
-                index += 1
-            else:
-                break
-        if (vlanID == ""):
+def logical_rows(out):
+    started_yielding = False
+    cur_row = []
+    for l in out.splitlines()[2:]:
+        if not l:
+            #"""Skip empty lines."""
```
Comment (webknjaz) [inline lib/ansible/modules/network/ios/ios_vlan.py]: Why did you comment a docstring?

File: lib/ansible/modules/network/ios/ios_vlan.py [hunk 10]
```diff
@@ -108,7 +108,6 @@
 
 
 def search_obj_in_list(vlan_id, lst):
-    obj = list()
```
Comment (webknjaz) [inline lib/ansible/modules/network/ios/ios_vlan.py]: This change is not related to the PR purpose, so it's better to send it as a separate PR.

*__Pro tip:__ It's the best practice to avoid mixing unrelated changes into the same patch. Make atomic logical changes relating to the very thing you want to change, do other changes in separate PRs.*

File: lib/ansible/modules/network/ios/ios_vlan.py [hunk 11]
```diff
@@ -211,38 +210,52 @@ def map_params_to_obj(module):
...
-
-        obj['interfaces'] = []
-        if len(splitted_line) > 3:
-            interface = []
-            for i in range(3, len(splitted_line)):
-                interface.append(splitted_line[i].replace('Gi', 'GigabitEthernet'))
-            obj['interfaces'].extend(interface)
-        objs.append(obj)
-
+def logical_rows(out):
```
Comment (webknjaz) [inline lib/ansible/modules/network/ios/ios_vlan.py]: It's not the most ideal code that I wrote, so if you feel you could improve things, especially function and variable naming — go for it :)

For example, function names normally designate actions and start with verbs (like `get_logical_rows()` or `extract_logical_rows()`), as opposed to variables, which are typically names of some things / nouns.

Keep in mind that I did a very quick sketch, so I might've missed some things, I rely on you here to check whether it fixes the issue and doesn't cause new problems.

File: lib/ansible/modules/network/ios/ios_vlan.py [hunk 12]
```diff
@@ -211,38 +210,52 @@ def map_params_to_obj(module):
     return obj
 
 
-def map_config_to_obj(module):
```
Comment (webknjaz) [inline lib/ansible/modules/network/ios/ios_vlan.py]: Also, I would probably want you to add some unit tests for the functions, which we introduced here as well as `map_config_to_obj(module)`, which we are changing in this PR.

File: lib/ansible/modules/network/ios/ios_vlan.py [hunk 13]
```diff
@@ -211,38 +211,49 @@ def map_params_to_obj(module):
...
+    obj = re.match(r'(?P<vlan_id>\d+)\s+(?P<name>[^\s]+)\s+(?P<state>[^\s]+)\s+(?P<interfaces>.*)', first_row).groupdict()
+    obj['interfaces'] = map_ports_str_to_list(obj['interfaces'])
+    obj['interfaces'].extend(prts_r for prts in rest_rows for prts_r in map_ports_str_to_list(prts))
+    return obj
+
+
+def parse_vlan_brief(vlan_out):
+    objs = []
+    for logical_rows in parse_to_logical_rows(vlan_out):
+        objs.append(parse_to_obj(logical_rows))
```
Comment (webknjaz) [inline lib/ansible/modules/network/ios/ios_vlan.py]: With this simplification, the pythonic way of generating a list is to use list comprehension. You can replace the whole function body with:
```python
return [parse_to_obj(r) for r in parse_to_logical_rows(vlan_out)]
```

File: lib/ansible/modules/network/ios/ios_vlan.py [hunk 14]
```diff
@@ -211,38 +211,49 @@ def map_params_to_obj(module):
...
+    rest_rows = logical_rows[1:]
+    obj = re.match(r'(?P<vlan_id>\d+)\s+(?P<name>[^\s]+)\s+(?P<state>[^\s]+)\s+(?P<interfaces>.*)', first_row).groupdict()
+    obj['interfaces'] = map_ports_str_to_list(obj['interfaces'])
+    obj['interfaces'].extend(prts_r for prts in rest_rows for prts_r in map_ports_str_to_list(prts))
+    return obj
+
+
+def parse_vlan_brief(vlan_out):
+    objs = []
+    for logical_rows in parse_to_logical_rows(vlan_out):
```
Comment (webknjaz) [inline lib/ansible/modules/network/ios/ios_vlan.py]: This name is misleading, because it's plural. But in fact `parse_to_logical_rows()` is a generator, which emits exactly one logical row per loop iteration. So if we were to keep this explicit loop it should've been renamed to `logical_row` or just `row` (because the generator name already contains a good explanation for what it emits).
But I'd go for just a list comprehension as stated in my comment one line below :)
It's kinda idiomatic Python way.

File: test/units/modules/network/ios/test_ios_vlan.py [hunk 15]
```diff
@@ -0,0 +1,77 @@
...
+
+    def test_parse_vlan_brief(self):
+        result = parse_vlan_brief(load_fixture('ios_vlan_config.cfg'))
+        obj = [
+            {
+                'name': 'default',
+                'interfaces': [
+                    'GigabitEthernet1/0/4',
+                    'GigabitEthernet1/0/5',
+                    'GigabitEthernet1/0/52'
```
Comment (webknjaz) [inline test/units/modules/network/ios/test_ios_vlan.py]: It's a good practice to have a trailing comma after the last sequence item as well. This way, when someone will add or remove an item it will generate only one line of diff, as opposed to two lines: one for the logical change and one for editing comma next to unrelated item.
This practice makes doing reviews easier and more joyful :)

File: test/units/modules/network/ios/test_ios_vlan.py [hunk 16]
```diff
@@ -0,0 +1,77 @@
...
+        self.load_config = self.mock_load_config.start()
+
+    def tearDown(self):
+        super(TestIosVlanModule, self).tearDown()
+        self.mock_run_commands.stop()
+        self.mock_load_config.stop()
+
+    def load_fixtures(self, commands=None, transport='cli'):
+        self.run_commands.return_value = load_fixture('ios_vlan_config.cfg')
+        self.load_config.return_value = dict(diff=None, session='session')
```
Comment (webknjaz) [inline test/units/modules/network/ios/test_ios_vlan.py]: please use dict literal instead of constructor:
```python
{'diff': None, 'session': 'session'}
```

File: lib/ansible/modules/network/ios/ios_vlan.py (hunk 9 above)
Comment (pierremahot) [reply lib/ansible/modules/network/ios/ios_vlan.py]: I don't know a lot I don't know it was a docstring

**Comment by pierremahot** [comment]
@webknjaz i lost my time on this error [commit](https://github.com/ansible/ansible/pull/40145/commits/29c3d499de1fafbef485f44f48b5edb874d5a414)

**Comment by pierremahot** [comment]
@webknjaz by implementing the testing i detected this error. Do I have to do a different pull request ?

File: lib/ansible/modules/network/ios/ios_vlan.py [hunk 17]
```diff
@@ -181,7 +181,7 @@ def map_obj_to_commands(updates, module):
         for h in have:
             obj_in_want = search_obj_in_list(h['vlan_id'], want)
             if not obj_in_want and h['vlan_id'] != '1':
-                commands.append('no vlan {0}'.format(h['vlan_id']))
+                commands.append('no vlan {}'.format(h['vlan_id']))
```
Comment (webknjaz) [inline lib/ansible/modules/network/ios/ios_vlan.py]: No, it's not that. If was asking you to remove this line: https://github.com/pierremahot/ansible/blob/6c12bc81cf45fab9f3f9940bf6b3cd73bb1396d8/test/sanity/pylint/ignore.txt#L50

File: lib/ansible/modules/network/ios/ios_vlan.py [hunk 18]
```diff
@@ -129,53 +129,53 @@ def map_obj_to_commands(updates, module):
 
         if state == 'absent':
             if obj_in_have:
-                commands.append('no vlan {}'.format(vlan_id))
+                commands.append('no vlan {0}'.format(vlan_id))
```
Comment (webknjaz) [inline lib/ansible/modules/network/ios/ios_vlan.py:132]: Why did you do this unrelated change? Do you want to support Python 2.6?

Comment (pierremahot) [reply lib/ansible/modules/network/ios/ios_vlan.py:132]: Don't know I just want to check all the box but I don't know what is good or not

Comment (webknjaz) [reply lib/ansible/modules/network/ios/ios_vlan.py:132]: What do you mean by checking box?

**Comment by webknjaz** [comment]
@pierremahot this PR looks okay to me now, it passes all the tests.

**Review by webknjaz — APPROVED** [review]

**Comment by pierremahot** [comment]
@webknjaz :smiley: thanks for your big help and I was thinking too improve testing but if it's ok for you then it's ok for me

**Comment by webknjaz** [comment]
@pierremahot well, I'm not an expert in this ios area, I was only helping with coding part from Software Engineering perspective. Of course there's lots of things we could refactor, but if we get fanatic about this it will be impossible to complete it in the nearest time and also it's not a good practice to be messing with different changes in same PR.

I'm going to ping some of networking ppl to take a look at this.

**Comment by pierremahot** [comment]
Ok I understand thanks again

File: lib/ansible/modules/network/ios/ios_vlan.py (hunk 18 above)
Comment (pierremahot) [reply lib/ansible/modules/network/ios/ios_vlan.py:132]: I don't know a lot about testing but by default the python 2.6 compatibility is checked by shippable but i don't know if it's needed. By checking box I mean to have all checks of shippable successful.

File: lib/ansible/modules/network/ios/ios_vlan.py [hunk 19]
```diff
@@ -211,36 +211,46 @@ def map_params_to_obj(module):
...
-        obj['interfaces'] = []
-        if len(splitted_line) > 3:
-            interface = []
-            for i in range(3, len(splitted_line)):
-                interface.append(splitted_line[i].replace('Gi', 'GigabitEthernet'))
-            obj['interfaces'].extend(interface)
-        objs.append(obj)
-
-    return objs
+    return parse_vlan_brief(run_commands(module, ['show vlan brief']))
```
Comment (Qalthos) [inline lib/ansible/modules/network/ios/ios_vlan.py]: Needs to be `run_commands(module, ['show vlan brief'])[0]`.

File: lib/ansible/modules/network/ios/ios_vlan.py [hunk 20]
```diff
@@ -211,36 +211,46 @@ def map_params_to_obj(module):
...
+
+
+def map_ports_str_to_list(ports_str):
+    return list(filter(bool, (p.strip().replace('Gi', 'GigabitEthernet') for p in ports_str.split(', '))))
+
+
+def parse_to_obj(logical_rows):
+    first_row = logical_rows[0]
+    rest_rows = logical_rows[1:]
+    obj = re.match(r'(?P<vlan_id>\d+)\s+(?P<name>[^\s]+)\s+(?P<state>[^\s]+)\s+(?P<interfaces>.*)', first_row).groupdict()
```
Comment (Qalthos) [inline lib/ansible/modules/network/ios/ios_vlan.py]: Your regex isn't properly capturing the case where there are no interfaces and also no trailing space at the end of the definition. Consider ending with `\s*(?P<interfaces>.*)`

**Review by Qalthos — CHANGES_REQUESTED** [review]

File: lib/ansible/modules/network/ios/ios_vlan.py (hunk 19 above)
Comment (webknjaz) [reply lib/ansible/modules/network/ios/ios_vlan.py]: why?

File: lib/ansible/modules/network/ios/ios_vlan.py (hunk 20 above)
Comment (webknjaz) [reply lib/ansible/modules/network/ios/ios_vlan.py]: Good catch!

File: lib/ansible/modules/network/ios/ios_vlan.py (hunk 19 above)
Comment (webknjaz) [reply lib/ansible/modules/network/ios/ios_vlan.py]: Oh, I see. `run_commands()` runs list of commands and returns list of results.

File: lib/ansible/modules/network/ios/ios_vlan.py (hunk 20 above)
Comment (webknjaz) [reply lib/ansible/modules/network/ios/ios_vlan.py]: @pierremahot we'll need a test for this

**Review by Qalthos — APPROVED** [review]

**Comment by jmagudo** [comment]
The fix introduces one minor bug. The parser in parse_to_logical_rows(out) ignores vlans starting with number 9.

The line 221:
        if '0' < l[0] < '9':
Perhaps should read:
        if '0' < l[0] <= '9':

**Comment by webknjaz** [comment]
@jmagudo Good catch! I would probably even do `if l[0].isdigit(): ` now
